src/@types/common.ts

```diff
@@ -20,3 +20,8 @@ export type NonEmptyArray<T> = [T, ...T[]];
 export type Without<T, U> = { [P in Exclude<keyof T, keyof U>]?: never };
 export type XOR<T, U> = T | U extends object ? (Without<T, U> & U) | (Without<U, T> & T) : T | U;
 export type Writeable<T> = { -readonly [P in keyof T]: T[P] };
+
+// Based on https://stackoverflow.com/a/57862073
```

Comment:
very good. What does it actually do? How do I use it? Please don't make me reverse-engineer typescript.